src/status_im/native_module/core.cljs

```diff
@@ -278,6 +279,52 @@
              :connection-string connection-string})
   (.inputConnectionStringForBootstrapping ^js (status) connection-string config-json callback))
 
+(defn deserialize-and-compress-key
```

Comment:
A couple things about this function:

1. If the dosctring says the function receives a `community id public key`, why is the name of the parameter called `desktop-key`?
2. Why is the docstring strictly speaking of a community ID, but the function name is generic? It seems the function works with any key, not just "public community ID keys". If this is the case, then we should try to keep the function agnostic.
3. The log `:fn` doesn't match the actual name of the function.

> gets back a key that status-mobile understands

This seems unnecessary. Everything developed in and for mobile should be understandable for mobile.

Suggestion for change (I removed the docstring because the function is clear enough without one):

```clojure
(defn deserialize-and-compress-key
  [key callback]
  (log/info "[native-module] Compressing public key"
            {:fn          :deserialize-and-compress-key
             :desktop-key key})
  (.deserializeAndCompressKey ^js (status) key callback))
```

Reply:
> This seems unnecessary. Everything developed in and for mobile should be understandable for mobile.

@ilmotta 
Hmm I wanted to be specific here because previously in certain scenarios we'd have community keys in chat which status-mobile would not understand and hence the "community not found" errors would popup. This PR was made to fix that.. and hence the docstring.. 
Does this make sense now?

Reply:
I understand the reasoning @siddarthkay, but to me this is still complecting domain-related stuff in a generic implementation. This is truly a utility and wrapper function, agnostic of domain.

> previously in certain scenarios we'd have community keys in chat which status-mobile would not understand and hence the "community not found" errors would popup.

The onus should be on the calling side. This function is agnostic of which key it receives, so the caller is responsible for using it appropriately, handling errors, etc.

---

I'm still confused by the parameter name `desktop-key`. What's a desktop key? Could it be just `key`?

Reply:
well.. naming variables / parameters is indeed one the things i suck at :)
my thought process behind keeping the parameter `desktop-key` was to remind me that here goes the community-id (key) we get from desktop and in callback we would get the key that mobile understands. ( for me .. `mobile-key`)
But as I understand this causes confusion amongst other devs who do not know the context, I'll be happy to rename it to be just `key`

Reply:
I see your point. It might be worth a shot adding these details to the docstring, but it's sure difficult for people without context to do this connection, given how generic this function is.

Reply:
`(defn deserialize-and-compress-key
  "Provides a community id (public key) to status-go which is first deserialized
  and then compressed. Example input/output :
  input key  = zQ3shTAten2v9CwyQD1Kc7VXAqNPDcHZAMsfbLHCZEx6nFqk9 and
  output key = 0x025596a7ff87da36860a84b0908191ce60a504afc94aac93c1abd774f182967ce6"
  [key callback]
  (log/info "[native-module] Deserializing and then compressing public key"
            {:fn  :deserialize-and-compress-key
             :key key})
  (.deserializeAndCompressKey ^js (status) key callback))`
  
  @ilmotta : this better?

Reply:
For sure @siddarthkay :) Thanks! Don't kill me, but if you want to go the extra mile a test is like *executable documentation*, so we could have a simple test describing the input/output.

Reply:
A test for this exists in status-go and we are just calling this method in status-mobile :) 
https://github.com/status-im/status-go/pull/3042/files#diff-79a6d848d7e638385a6ebc36d289250f869979e18cd17c9c1ef810da690836cfL27

Reply:
Sounds good, given this function is just a wrapper. Thanks!